ui/src/app/core/auth.service.ts

```diff
@@ -96,7 +115,47 @@ export class AuthService {
     return auth2.signOut();
   }
 
+  private revokeToken(): Promise<any> {
+    const auth2 = gapi.auth2.getAuthInstance();
+    return auth2.disconnect();
+  }
+
   private handleError(error): void {
       this.snackBar.open('An error occurred: ' + error);
   }
+
+  private setUpEventListeners(): void {
+    const mouseWheelStream = Observable.fromEvent(window, "mousewheel");
+    mouseWheelStream.subscribe(() => this.resetTimers());
+
+    const mouseDownStream = Observable.fromEvent(window, "mousedown");
+    mouseDownStream.subscribe(() => this.resetTimers());
+
+    const mouseMoveStream = Observable.fromEvent(window, "mousemove");
+    mouseMoveStream.subscribe(() => this.resetTimers());
+
+    const keyDownStream = Observable.fromEvent(window, "keydown");
+    keyDownStream.subscribe(() => this.resetTimers());
+
+    const keyUpStream = Observable.fromEvent(window, "keyup");
+    keyUpStream.subscribe(() => this.resetTimers());
+
+    this.resetTimers();
+  }
+
+  public resetTimers(): void {
+    window.clearTimeout(this.logoutTimer);
+    window.clearTimeout(this.warningTimer);
+    this.snackBar.dismiss();
+
+    this.warningTimer = window.setTimeout(() => {
+      this.snackBar.open('You are about to be logged out due to inactivity');
```

Comment:
I think it would be great to tell the user that they have 10 seconds in this message - it's pretty short compared to the one minute or 30 seconds I've seen elsewhere

Reply:
I think having the static value of "10 seconds" will be confusing to users.  What if they only spot the message when they actually have only 5 seconds to go?  Unfortunately, I haven't found a way to easily update the message in the `SnackBar` component without closing it and opening it again, so I don't know how to show an actual countdown.